### PR TITLE
Use Ruby 3

### LINKED ISSUE #40: Tests for Word Count pass locally with Ruby 2.6 but fail on Exercism website because test runner is using Ruby 2.5.9
In the Word Count exercise, I'm using `to_h` with a block, which only becomes available in Ruby 2.6. 
https://rubyreferences.github.io/rubychanges/2.6.html#to_h-with-a-block

The error shows for this line in my code: 
`words.to_h {|word| [word, words.count(word)] }`

> TypeError: wrong element type String at 0 (expected array)
> 
> Traceback (most recent call first):
>     Line 10:in 'to_h'
>     Line 10:in 'word_count'

I found the Ruby version by adding this in the code editor:
`"ruby #{ RUBY_VERSION }p#{ RUBY_PATCHLEVEL }".frog`
> NoMethodError: undefined method `frog' for "ruby 2.5.9p229":String

[After that, I discovered there's `debug` method! 😄 ]

I came here from: https://exercism.org/docs/using/solving-exercises/tests-pass-locally 
although I see that the code in this repo already has a .ruby-version of ruby-2.6.6. 

I also see that https://github.com/exercism/ruby says
> You'll need a recent (2.6+) version of Ruby

So I assume at some point the test runner on the website will be updated to 2.6?
Not sure if this is the right place to report this. Thanks! 😊

### PR DESCRIPTION
**Pull request by iHiD**
Closes https://github.com/exercism/ruby-test-runner/issues/40

### CODE REVIEW

**Review by ErikSchierboom — APPROVED** [review]
Is Ruby 3 now the minimum version, or is it backwards compatible?

**Comment by iHiD** [comment]
It's pretty backwards compatible. Code that raised deprecation warnings in Ruby 2.7 will now result in ArgumentError or different behaviour, but this is mainly around where kwargs and the final args being hashes could be confusing.

As Ruby 3.0 is now "normal" in production, I think we should probably make the upgrade.

**Comment by ErikSchierboom** [comment]
:+1

**Comment by Steffan153** [comment]
When is this going to be merged? :)

**Comment by iHiD** [comment]
The test runner codebase isn't actually compatible with Ruby 3 so that needs fixing first, then it'll get merged :)

**Comment by kotp** [comment]
We could mark this as draft until the prerequisite is done.  That will prevent it from coming in early.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:2.5-alpine
+FROM ruby:3.02-alpine
```
Comment (katrinleinweber) [inline Dockerfile:1]: Is that a typo? On my machine, `docker pull` on that yields: 

> failed to solve with frontend dockerfile.v0: failed to create LLB definition: docker.io/library/ruby:3.02-alpine: not found

But this builds:

```suggestion
FROM ruby:3.0.2-alpine
```

**Comment by katrinleinweber** [comment]
> … codebase isn't actually compatible with Ruby 3 …

@iHiD: Are there specific lines that need to be changed? I tried relying on RubyMine's default inspections & "problem" detection just now, but found none.

I'd be interested in taking a stab at this task, but I'd need a few pointers and help to get started.

---

Specifically, I'm encountering this after following the [README's instructions](https://github.com/exercism/ruby-test-runner/blob/9345867127a29b3aebcdb4b9d077a4b5f01bb9a1/README.md#running-the-tests):

> Minitest::UnexpectedError: Errno::ENOENT: No such file or directory @ rb_sysopen - /tmp/…uuid…/output/results.json

for all [TestRunnerTest#… tests](https://github.com/exercism/ruby-test-runner/blob/9345867127a29b3aebcdb4b9d077a4b5f01bb9a1/test/test_runner_test.rb#L4-L131), and for [AttacksTest#test_large_output_is_truncated](https://github.com/exercism/ruby-test-runner/blob/9345867127a29b3aebcdb4b9d077a4b5f01bb9a1/test/attacks_test.rb#L4-L21).

The folders are being created just fine, the `input/…/*.rb` files as well, so it seems that somewhere, _writing_ the test results JSON fails (silently, it also seems).

File: Dockerfile (hunk 1 above)
Comment (iHiD) [reply Dockerfile:1]: That seems more correct.

**Comment by iHiD** [comment]
@katrinleinweber Hi there :) Thanks for picking this up.

Try uncommenting one of https://github.com/exercism/ruby-test-runner/blob/main/test/test_helper.rb#L61 and commenting out the ones below. You should get better info then. This is a weird project as it's effectively wrapping minitest in an external runner, and capturing stdout/stderr, so getting logging out can be tricky. But that should help.

I think the issue might have been that Mandate wasn't compatible with Ruby 3, but try using `1.0.0.beta1` for that and see if it fixes it. (I maintain Mandate). Any/most pain is probably around kwargs and similar, which is where Ruby 3 changes in quite a breaking way.

Happy to answer questions and I can take a look at things if there's anything specific that needs another pair of eyes.

File: Dockerfile (hunk 1 above)
Comment (katrinleinweber) [reply Dockerfile:1]: Done in https://github.com/exercism/ruby-test-runner/pull/47

**Comment by katrinleinweber** [comment]
Thank you @iHiD! I'm suggesting https://github.com/exercism/ruby-test-runner/pull/47 as an alternative to this PR.

**Comment by iHiD** [comment]
Closing in favour of #47

**Pull request closed without merging**